src/CodeReview.Orchestrator/Commands/ExtractMetadataCommand.cs

```diff
@@ -9,43 +9,34 @@ namespace GodelTech.CodeReview.Orchestrator.Commands
 {
     public class ExtractMetadataCommand : IExtractMetadataCommand
     {
-        private static readonly Type[] Options =
-        {
-            typeof(RunOptions),
-            typeof(EvaluateOptions),
-            typeof(ExtractMetadataOptions),
-            typeof(NewAnalysisManifestOptions),
-            typeof(NewDockerEngineCollectionManifestOptions)
-        };
-
         private readonly IFileService _fileService;
         private readonly IPathService _pathService;
         private readonly IDirectoryService _directoryService;
-        private readonly ICommandLineJsonConvert _commandLineJsonConvert;
+        private readonly IOptionMetadataProvider _optionMetadataProvider;
 
         public ExtractMetadataCommand(
             IFileService fileService, 
             IPathService pathService, 
             IDirectoryService directoryService, 
-            ICommandLineJsonConvert commandLineJsonConvert)
+            IOptionMetadataProvider optionMetadataProvider)
         {
             _fileService = fileService ?? throw new ArgumentNullException(nameof(fileService));
             _pathService = pathService ?? throw new ArgumentNullException(nameof(pathService));
             _directoryService = directoryService ?? throw new ArgumentNullException(nameof(directoryService));
-            _commandLineJsonConvert = commandLineJsonConvert ?? throw new ArgumentNullException(nameof(commandLineJsonConvert));
-        }
+            _optionMetadataProvider = optionMetadataProvider ?? throw new ArgumentNullException(nameof(optionMetadataProvider));
+        } 
 
         public async Task<int> ExecuteAsync(ExtractMetadataOptions options)
         {
             if (options is null) throw new ArgumentNullException(nameof(options));
-            if (string.IsNullOrWhiteSpace(options.OutputPath)) throw new ArgumentException("Value cannot be null or whitespace.", nameof(options.OutputPath));
-
-            var jsonString = _commandLineJsonConvert.Serialize(Options);
+            if (string.IsNullOrWhiteSpace(options.OutputPath)) throw new ArgumentNullException(nameof(options.OutputPath), "Value cannot be null or whitespace.");
 
+            var metadata = _optionMetadataProvider.GetOptionsMetadata();
             var directoryPath = _pathService.GetDirectoryName(options.OutputPath);
+            
             _directoryService.CreateDirectory(directoryPath);
```

Comment:
I'm wondering if we need to check if folder exists. Will this code work if folder already exists?

Reply:
yup, it will

https://docs.microsoft.com/en-us/dotnet/api/system.io.directory.createdirectory?view=net-5.0
```
If the directory already exists, this method does not create a new directory, but it returns a DirectoryInfo object for the existing directory.
```